stories/modal.stories.js

```diff
@@ -0,0 +1,91 @@
+import React from 'react';
+import ReactModal from 'react-modal';
+import { storiesOf } from '@storybook/react';
+import { number, text, withKnobs } from '@storybook/addon-knobs';
+
+import { Box, Button, Card, Grid, Heading, Icon, Modal } from '../src';
+
+// For modal accessibility. See http://reactcommunity.org/react-modal/accessibility/ for details
+ReactModal.setAppElement(document.body); // eslint-disable-line no-undef
+
+class ModalStory extends React.Component {
+  state = {
+    modalIsOpen: true
+  };
+
+  closeModal = () => {
```

Comment:
Any thoughts on if it's more appropriate for the modal to manage its state or the parent to manage its state?

Reply:
My vote is to leave it as controlled (changing modal state via prop).
If we want an uncontrolled one down the road, we could always create a wrapper.

Reply:
Whoops sorry @mikecx saw the approval before I saw your Q. I did think about that a bit. react-modal handles the dispaly of the modal via an isOpen prop. I started playing with the idea of wrapping it with our own modal component that managed state but in most (all?) cases you would be opening the modal through some interaction or event. The only way I could see it managing its own state is if it was open by default, then you could close it. However in that situation you wouldn't be able to open it again. Did you have a scenario in mind?